BCP195bis/draft-ietf-uta-rfc7525bis.md

```diff
@@ -341,6 +368,27 @@ Rationale: SNI supports deployment of multiple TLS-protected virtual servers on
       target domain for a given connection; this information leak will be plugged by 
       use of TLS Encrypted Client Hello.
 
+In order to prevent the attacks described in {{ALPACA}}, a receiver who does
+not recognize the presented server name SHOULD NOT continue the handshake and
+instead fail with a fatal-level `unrecognized_name(112)` alert.  Note that this
+recommendation updates Section 3 of {{!RFC6066}}: "If the server understood the
+ClientHello extension but does not recognize the server name, the server SHOULD
+take one of two actions: either abort the handshake by sending a fatal-level
+`unrecognized_name(112)` alert or continue the handshake."
+
+## Application-Layer Protocol Negotiation
+
+TLS implementation MUST support the Application-Layer Protocol Negotiation
+(ALPN) extension {{!RFC7301}}.
+
```

Comment:
We should add at least SHOULD on the client side. Maybe MUST because it doesn't have the security downsides of SNI.

Reply:
Is the current text (i.e., "TLS implementation MUST [...]" irrespective of the role) ok then?

Reply:
"Irrespective of the role" is good. Or else: TLS implementations (both client- and server-side).

Reply:
done in 8b6e6b2